MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+MIT License
+
+Copyright (c) 2019 Gérald Barré
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+#nullable enable
+#pragma warning disable CS1591
+#pragma warning disable CA1034
+using System;
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+
+namespace MediaBrowser.Common.Extensions
+{
+    /// <summary>
+    /// Extension class for splitting lines without unnecessary allocations.
+    /// </summary>
+    public static class SplitLinesStringExtensions
+    {
+        /// <summary>
+        /// Creates a new line split enumerator.
+        /// </summary>
+        /// <param name="str">The string to split.</param>
+        /// <param name="separator">The separator to split on.</param>
+        /// <returns>The enumerator struct.</returns>
+        [Pure]
+        public static LineSplitEnumerator SpanSplit(this string str, char separator) => new (str.AsSpan(), separator);
+
+        /// <summary>
+        /// Creates a new line split enumerator.
+        /// </summary>
+        /// <param name="str">The span to split.</param>
+        /// <param name="separator">The separator to split on.</param>
+        /// <returns>The enumerator struct.</returns>
+        [Pure]
+        public static LineSplitEnumerator Split(this ReadOnlySpan<char> str, char separator) => new (str, separator);
+
+        [StructLayout(LayoutKind.Auto)]
+        public ref struct LineSplitEnumerator
```

Comment:
This isn't a `LineSplitEnumerator` anymore

Reply:
True. I'm terrible with names though

Reply:
Just `SplitEnumerator`?